src/main/java/org/duke/cmd/Command.java

```diff
@@ -0,0 +1,69 @@
+package org.duke.cmd;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * Represents parsed user input, divided into a command type and arguments.
+ */
+public class Command {
+    private static final Pattern switchBoundary = Pattern.compile("\\s+/");
+    private static final Pattern cmdBoundary = Pattern.compile("\\s+");
+    public final String type;
+    public final String arguments;
+    public final Map<String, String> namedArguments;
+    public Command(String type, String arguments, Map<String, String> namedArguments) {
```

Comment:
I felt that the access modifiers for the above fields could be set to private to prevent its values from being modified freely by other classes.

Reply:
The `Command` class is intended to be an immutable value type, akin to a C struct. The fields cannot be modified, as they are final (bar the `Map` which has interior mutability). You still have a point in that it breaks encapsulation, so future updates may break consumers of the API.

At the very least, I'll fix it up for the Map case by wrapping it with `Collections.unmodifiableMap()`. Thanks for the catch.